File: src/main/java/com/example/kdp/entity/Product.java
package com.example.kdp.entity;

import java.util.Date;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;

@Entity
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int productId;
    private String productName;
    private double sellingPrice;
    private double originalPrice;
    private Date postedDate;
    private String companyName;


    public Product() {
    }

    public Product(
        String productName,
        double sellingPrice,
        double originalPrice,
        Date postedDate,
        String companyName) {

        this.productName = productName;
        this.sellingPrice = sellingPrice;
        this.originalPrice = originalPrice;
        this.postedDate = postedDate;
        this.companyName = companyName;
    }
    // Setters

    public void setProductId(int productId) {
        this.productId = productId;
    }
    public void setProductName(String productName) {
        this.productName = productName;
    }

    public void setSellingPrice(double sellingPrice) {
        this.sellingPrice = sellingPrice;
    }

    public void setOriginalPrice(double originalPrice) {
        this.originalPrice = originalPrice;
    }

    public void setPostedDate(Date postedDate) {
        this.postedDate = postedDate;
    }

    public void setCompanyName(String companyName) {
        this.companyName = companyName;
    }

    // Getters

    public int getProductId() {
        return productId;
    }

    public String getProductName() {
        return productName;
    }

    public double getSellingPrice() {
        return sellingPrice;
    }

    public double getOriginalPrice() {
        return originalPrice;
    }

    public Date getPostedDate() {
        return postedDate;
    }

    public String getCompanyName() {
        return companyName;
    }
}

File: frontend/src/types/Product.ts

// "productId": 1,
//     "productName": "Oppo Find X8",
//     "companyName": "Oppo",
//     "sellingPrice": 61999
export interface Products {

        productId : number;
        productName : string;
        companyName : string;
        sellingPrice : number;
}

export interface Product {
      
      productId : number;
      productName : string;
      companyName : string;
      sellingPrice : number;
      originalPrice : number;
      postedDate : string;
}

export interface ProductState {

       products : Products[];
       singleProduct : Product | null;
       loading : boolean;
       error : string  | null;

}